Cargo.toml

```
@@ -35,10 +35,19 @@ rmp-serde = { version = ">=1", default-features = false, optional = true }
 trybuild = { version = "1.0", features = ["diff"] }
 
 [features]
-default = ["derive"]
+default = ["std", "derive"]
+
+# Provide impls for common standard library types like Vec<T> and HashMap<K, V>.
+# Requires a dependency on the Rust standard library.
+std = ["alloc"]
+
+# Provide impls for types in the Rust core allocation and collections library
+# including String, Box<T>, Vec<T>, and Cow<T>. This is a subset of std but may
+# be enabled without depending on all of std.
+alloc = []
 
 # Provides a `derive` macro for implementing the `Transient` trait
-derive = ["transient-derive"]
+derive = ["std", "transient-derive"]
```

The derive macro does not rely on `std`, so this shouldn't be needed.
```suggestion
derive = ["transient-derive"]
```

---

My logic on adding that was mostly to communicate that `transient-derive` (and proc-macro crates in general by my understanding) is *not* `no_std`, and that implicitly pulling it in by leaving the `derive` feature active will require `std`. So disabling the "std" feature but keeping the "derive" feature would be pretty pointless.

However, I see that Serde follows your logic instead of mine, so maybe that is the way to go. What do you think?

---

Executing the derive macro requires std *at build time*. Proc-macro crates do not implicitly add a dependency on std. A `no_std` crate can still use derive macros, as long as the generated code doesn't use anything from std.

---

Wow, I have never had to deal with `no_std` environments in my work and didn't even realize you could use derives in them! Very good to know, thanks for the review!